### ISSUE TITLE
Adding CHANGELOG.md to the lein new templates

### ISSUE THREAD

**Issue by danielcompton** [technomancy/leiningen#1982]
What are your thoughts about adding a CHANGELOG.md file to the lein new template with a skeleton frame setup that follows the patterns recommended by http://keepachangelog.com? It would be similar to the skeleton docs and tests that are created. If people don't want it they can delete it easily, but it would guide people towards good practices. 

I'm happy to do the PR but wanted feedback on this first.

**Comment by hyPiRion** [comment]
I think this is good practice, although I'm on the fence on whether I want to include this or not. Too much stuff in the templates may be off-putting to people new to Clojure.

Do you have any thoughts on this @technomancy?

**Comment by technomancy** [comment]
I'm for it. Most projects delete the `doc/` directory we create, but we should be setting a high standard for proper practices; people who don't like it can delete the files.

**Comment by mbertheau** [comment]
Maybe then documentation should be added for which files are required for correct operation and which aren't.

**Comment by danielcompton** [comment]
@mbertheau I think it should be pretty self evident which files are required for the user? What do you think?

**Comment by mbertheau** [comment]
@danielcompton It is only when you know it. A week ago I was new and I would have liked an overview about what the function of each file in the template is.

Imho Leiningen is missing a short high level explanation and reference documentation for the tasks. Or these exist but I didn't find them. I found something very useful now in this http://www.flyingmachinestudios.com/programming/how-clojure-babies-are-made-what-leiningen-is/ and related blog posts, and apparently braveclojure.com has some good material as well. Before I found that though, using Leiningen was quite frustrating. I didn't know what was going on, an while everything worked that was fine, but when something broke, I had no idea where to start.
